### PR TITLE
Fix for storedPaymentSources raising exception

### PR DESCRIPTION
**Pull request by kswiatek92**
Closes [#4907](https://github.com/mirumee/saleor/issues/4907)

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 8f971bc47d6a44be2cde7ff9823f891320cf1498 (kswiatek92/saleor @ fix/field-storedPaymentSopurces-raises-exception)
Base comparison is 6d5ce07e9f8faa3c0f4b8789cb4945157cb59940.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         14	         14	              0
  complete checkout                          	          6	          6	              0
  create checkout                            	         48	         48	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	         13	         13	              2

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/graphql/account/resolvers.py [hunk 1]
```diff
@@ -106,7 +106,8 @@ def resolve_address_validation_rules(
 
 def resolve_payment_sources(user: models.User):
     stored_customer_accounts = (
-        (gtw, fetch_customer_id(user, gtw)) for gtw in gateway.list_gateways()
+        (gtw["name"], fetch_customer_id(user, gtw["name"]))
```
Comment (korycins) [inline saleor/graphql/account/resolvers.py]: Let's add some tests for this account.

Comment (kswiatek92) [reply saleor/graphql/account/resolvers.py]: added query test, and modify payment test so it doesn't mock `list_gateways()`

File: tests/api/test_payment.py [hunk 2]
```diff
@@ -541,8 +546,15 @@ def set_braintree_customer_id(customer_user, braintree_customer_id):
     return customer_user
 
 
+@pytest.fixture
+def set_dummy_customer_id(customer_user, dummy_customer_id):
```
Comment (korycins) [inline tests/api/test_payment.py]: Where do you use it?

Comment (kswiatek92) [reply tests/api/test_payment.py]: `test_list_payment_sources`

`customer_user` is used in both `set_dummy_customer_id` and `user_api_client`
that's why I don't have to explicitly use this to retrieve customer_user

**Pull request merged**